chef/cookbooks/cpe_chrome/resources/cpe_chrome.rb

```diff
@@ -94,5 +96,31 @@ def manage_chrome_osx
         ],
       }
     end
+
+    unless node['cpe_chrome']['mp']['UseMasterPreferences'] == false
+      mprefs = node['cpe_chrome']['mp']['FileContents'].reject { |v| v.nil? }
```

Comment:
Ah okay, it looks like this yields a chef thing instead of a bare array:
```
chef:attributes (12.18.31)> default['cpe_chrome']['mp']['FileContents'].class
 => Chef::Node::AttrArray
```
You can call a `to_ary` on it, which yields an array and will allow the use of `compact` to get the behavior you want:
```
chef:attributes (12.18.31)> default['cpe_chrome']['mp']['FileContents'] = ['a', 'b', nil, nil, 'x']
 => ["a", "b", nil, nil, "x"]
chef:attributes (12.18.31)> default['cpe_chrome']['mp']['FileContents'].to_ary.compact
 => ["a", "b", "x"]
```